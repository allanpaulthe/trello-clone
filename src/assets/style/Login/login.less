.login-section{
    overflow: hidden;
    padding: 4em 1em;
    word-wrap: break-word;
    .login-wrapper{
        max-width: 430px;
        margin:0 auto;
    }
    h1{
        height: 100%;
        color: #333;
        font-size: 38px;
        line-height: 48px;
        margin-top: 0;
        margin-bottom: 0
    }
    @media screen and (max-width: 400px) {
        padding: 2em 1em;
    }
    @media screen and (max-width: 650px) {
        padding: 0 1em;
    }
    a{
        color: #298fca !important;
        font-size: 18px;
        outline: none;
        text-decoration: underline;
        &:hover{
            color: darken(#298fca,20%) !important;
        }
    }
    span{
        font-size: 18px;
    }
    a {
        color: #82bfe2;
    }
    .link>a {
        margin-top: 2.2em;
    }
}
