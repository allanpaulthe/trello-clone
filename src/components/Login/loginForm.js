import React, { Component } from 'react';
import '../../assets/style/Login/loginform.less';
import { withRouter } from 'react-router-dom';

class LoginForm extends Component {
    constructor(props) {
        super(props);
        this.state = {
            email: '',
            password: '',
            wrong: false
        };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }
    handleSubmit(event) {
        event.preventDefault();
        let email = this.state.email.toUpperCase();
        let pass = this.state.password.toUpperCase();
        if (email === 'ALAN@GMAIL.COM' && pass === 'ALAN') {
            window.location.pathname = '/user/boards';
        }
        else {
            this.setState({
                wrong: true
            });
        }
    }

    handleChange(event) {
        this.setState({
            [event.target.id]: event.target.value,
        });
    }

    render() {
        return (
            <form action="">
                <div className="loginForm">
                    {this.state.wrong &&
                        <p className="error">Wrong Credentials!!!..</p>
                    }
                    <p>Email <span>(or username)</span> </p>
                    <input type="text" placeholder="e.g., e.hughes@downton.org.uk"
                        id="email"
                        value={this.state.email}
                        onChange={this.handleChange}
                        required
                    />
                    <p>Password</p>
                    <input type="password" placeholder="e.g., ••••••••••••"
                        id="password"
                        value={this.state.passWord}
                        onChange={this.handleChange}
                        required
                    />
                    <input type="submit" value="Log In" className="button green-button"
                        onClick={this.handleSubmit}
                    />
                </div>
            </form>
        );
    }
}

export default withRouter(LoginForm);