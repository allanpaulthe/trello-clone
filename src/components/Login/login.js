import React, { Component } from 'react';
import Footer from '../Home/footer';
import GoogleButtton from '../Home/GoogleButtton';
import '../../assets/style/Login/login.less';
import LoginForm from './loginForm';
import { BrowserRouter as Router } from 'react-router-dom';

class Login extends Component {
    constructor(props) {
        super(props);
        this.state = {
            emailAndPass: false
        };
        this.handleClick = this.handleClick.bind(this);
    }
    handleClick() {
        this.setState({
            emailAndPass: true
        });
    }
    render() {
        return (
            <Router>
                <div>
                    <section className="login-section">
                        <div className="login-wrapper">
                            <div>
                                <h1>Log in to Trello</h1>
                                <span>or <a>create an account</a></span>
                                <div>
                                    {this.state.emailAndPass &&
                                        <LoginForm />
                                    }
                                    <GoogleButtton />
                                    {!this.state.emailAndPass &&
                                        <p onClick={this.handleClick}> <a>Log in with email and password</a> </p>
                                    }
                                    <p> <a href="" tabindex="3">Log in with SSO</a> </p>
                                    {this.state.emailAndPass &&
                                        <p> <a>Forgot your password?</a> </p>
                                    }
                                </div>
                            </div>
                        </div>
                    </section>
                    <Footer />
                </div>
            </Router>
        );
    }
}

export default Login;