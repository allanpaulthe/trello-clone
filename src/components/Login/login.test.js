import React from 'react';
import ReactDom from 'react-dom';
import Login from './login';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('login', () => {
    it('renders correctly', () => {
        const output = shallow(
            <Login />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
    it('should handle the click event to display form', () => {
        const output = shallow(
            <Login />
        );
        expect(output.state().emailAndPass).toEqual(false)
        const link=output.find('p').get(0);
        link.props.onClick();
        expect(output.state().emailAndPass).toEqual(true);
    });

});