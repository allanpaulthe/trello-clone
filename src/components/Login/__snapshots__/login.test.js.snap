// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login renders correctly 1`] = `
<BrowserRouter>
  <div>
    <section
      className="login-section"
    >
      <div
        className="login-wrapper"
      >
        <div>
          <h1>
            Log in to Trello
          </h1>
          <span>
            or 
            <a>
              create an account
            </a>
          </span>
          <div>
            <GoogleButton />
            <p
              onClick={[Function]}
            >
               
              <a>
                Log in with email and password
              </a>
               
            </p>
            <p>
               
              <a
                href=""
                tabindex="3"
              >
                Log in with SSO
              </a>
               
            </p>
          </div>
        </div>
      </div>
    </section>
    <Footer />
  </div>
</BrowserRouter>
`;
